rclpy/test/test_action_client.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+import uuid
+
+import rclpy
+from rclpy.action_client import ActionClient
+from rclpy.executors import SingleThreadedExecutor
+
+from test_msgs.action import Fibonacci
+
+from unique_identifier_msgs.msg import UUID
+
+
+# TODO(jacobperron) Reduce fudge once wait_for_service uses node graph events
+TIME_FUDGE = 0.3
+
+
+class MockActionServer():
+
+    def __init__(self, node):
+        self.goal_srv = node.create_service(
+            Fibonacci.GoalRequestService, '/fibonacci/_action/send_goal', self.goal_callback)
+        self.cancel_srv = node.create_service(
+            Fibonacci.CancelGoalService, '/fibonacci/_action/cancel_goal', self.cancel_callback)
+        self.result_srv = node.create_service(
+            Fibonacci.GoalResultService, '/fibonacci/_action/get_result', self.result_callback)
+        self.feedback_pub = node.create_publisher(
+            Fibonacci.Feedback, '/fibonacci/_action/feedback')
+
+    def goal_callback(self, request, response):
+        response.accepted = True
+        return response
+
+    def cancel_callback(self, request, response):
+        response.goals_canceling.append(request.goal_info)
+        return response
+
+    def result_callback(self, request, response):
+        return response
+
+    def publish_feedback(self, goal_id):
+        feedback = Fibonacci.Feedback()
+        feedback.action_goal_id = goal_id
+        self.feedback_pub.publish(feedback)
+
+
+class TestActionClient(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.context = rclpy.context.Context()
+        rclpy.init(context=cls.context)
+        cls.executor = SingleThreadedExecutor(context=cls.context)
+        cls.node = rclpy.create_node('TestActionClient', context=cls.context)
+        cls.mock_action_server = MockActionServer(cls.node)
+        cls.feedback = None
```

Comment:
IIUC this sets feedback to `None` once before all tests are run, but the tests expect `feedback` cleared before each test is run. That can be done with a `setUp()` method instead.